### PR TITLE
Fixed the failing testcases; Moved the additional_columns processing …

### PR DESCRIPTION
**Pull request by Nivi09**
- Fixed the failing testcases.
- Moved the additional_columns processing logic from main.py to inference_postprocessor.py.

### CODE REVIEW

File: assets/aml-benchmark/components/src/aml_benchmark/inference_postprocessor/inference_postprocessor.py [hunk 1]
```diff
@@ -175,29 +175,24 @@ def read_ground_truth_dataset(self) -> pd.DataFrame:
...
                 ]
             if self.additional_columns:
-                elements = self.additional_columns.split(",")
-                strips = [s.strip() for s in elements if s.strip()]
-                for column in strips:
-                    try:
-                        result_df[column] = actual_df[column]
-                    except KeyError:
-                        raise BenchmarkUserException._with_error(
+                columns = [col.strip() for col in self.additional_columns.split(',')]
```
Comment (arun-rajora) [inline assets/aml-benchmark/components/src/aml_benchmark/inference_postprocessor/inference_postprocessor.py:185]: We also need to remove string that have become empty after stripping (like it was done before).

File: assets/aml-benchmark/components/src/aml_benchmark/inference_postprocessor/inference_postprocessor.py [hunk 2]
```diff
@@ -175,29 +175,24 @@ def read_ground_truth_dataset(self) -> pd.DataFrame:
...
+                try:
+                    result_df[columns] = actual_df[columns]
+                except KeyError as e:
+                    raise BenchmarkUserException._with_error(
                             AzureMLError.create(
                                 BenchmarkUserError,
-                                error_details=f"Column {column} doesn't exist.\
-                                    Please check your data before submitting again.")
+                                error_details=f"Error: {e}.Please check your data before submitting again.")
```
Comment (arun-rajora) [inline assets/aml-benchmark/components/src/aml_benchmark/inference_postprocessor/inference_postprocessor.py:192]: Can we make the error descriptive that says which column specified in additional_columns was not present in the dataframe.

**Review by arun-rajora — CHANGES_REQUESTED** [review]

File: assets/aml-benchmark/components/src/aml_benchmark/inference_postprocessor/inference_postprocessor.py (hunk 2 above)
Comment (Nivi09) [reply assets/aml-benchmark/components/src/aml_benchmark/inference_postprocessor/inference_postprocessor.py:192]: Will change as discussed. Will make the error message more descriptive.

File: assets/aml-benchmark/components/src/aml_benchmark/inference_postprocessor/inference_postprocessor.py (hunk 1 above)
Comment (Nivi09) [reply assets/aml-benchmark/components/src/aml_benchmark/inference_postprocessor/inference_postprocessor.py:185]: Will raise a warning letting user know that they have provided empty string in the additional columns and then drop those empty strings from additional columns.

**Review by iamrk04 — DISMISSED** [review]

**Review by iamrk04 — CHANGES_REQUESTED** [review]
the script_syntax workflow is failing. Could you please check that?

**Review by arun-rajora — APPROVED** [review]

**Pull request merged**